src/main/java/org/opensearch/agent/tools/RAGTool.java

```diff
@@ -151,25 +149,25 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
                     resultBuilder.append("_source: ").append(source.toString()).append("\n");
                 }
 
-                vectorDBToolOutput = (T) gson.toJson(resultBuilder.toString());
+                queryToolOutput = (T) gson.toJson(resultBuilder.toString());
             }
 
             Map<String, String> tmpParameters = new HashMap<>();
             tmpParameters.putAll(parameters);
 
-            if (vectorDBToolOutput instanceof List
-                && !((List) vectorDBToolOutput).isEmpty()
-                && ((List) vectorDBToolOutput).get(0) instanceof ModelTensors) {
-                ModelTensors tensors = (ModelTensors) ((List) vectorDBToolOutput).get(0);
+            if (queryToolOutput instanceof List
```

Comment:
The `queryToolOutput`  is always String, right? Why we want to check if it's List or ModelTensor?

Reply:
removed